test/test_client.py

```diff
@@ -474,6 +481,86 @@ def test_set_plc_datetime(self):
         # Can't actual set datetime in emulated PLC, get_plc_datetime always returns system time.
         # self.assertEqual(new_dt, self.client.get_plc_datetime())
 
+    def test_wait_as_completion_pass(self, timeout=10):
+        # Cli_WaitAsCompletion
+        # prepare Server with values
+        area = snap7.types.areas.DB
+        dbnumber = 1
+        size = 1
+        start = 1
+        data = bytearray(size)
+        self.client.write_area(area, dbnumber, start, data)
+        # start as_request and test
+        p_data = self.client.as_db_read(dbnumber, start, size)
+        try:
+            self.client.wait_as_completion(ctypes.c_ulong(timeout))
+        except Snap7Exception as s7_err:
+            if s7_err.args[0] == b'CLI : Job Timeout':
+                self.fail(f"Request was timeouted after {timeout} seconds - FAIL")
+            else:
+                self.fail(f"Snap7Exception was thrown: {s7_err} - FAIL")
+        except BaseException as pyt_err:
+            self.fail(f"Exception was thrown: {pyt_err}")
+        self.assertEqual(bytearray(p_data), data)
+
+    def test_wait_as_completion_timeouted(self, timeout=0, tries=500):
+        # Cli_WaitAsCompletion
+        # prepare Server
+        area = snap7.types.areas.DB
+        dbnumber = 1
+        size = 1
+        start = 1
+        data = bytearray(size)
+        self.client.write_area(area, dbnumber, start, data)
+        # start as_request and wait for zero seconds to try trigger timeout
+        for i in range(tries):
+            self.client.as_db_read(dbnumber, start, size)
+            try:
+                self.client.wait_as_completion(ctypes.c_ulong(0))
```

Comment:
The method itself is raising an Snap7Exception in case of a timeout, which is the original behaviour of the snap7 version as well.
The goal  of this test is to catch this Snap7Exception` "CLI : Job Timeout" ` so that the behaviour of `wait_as_completion(...)` in case of a Job Timeout can be tested. It is possible that other Snap7Exceptions may be raised, so these have to be caught and checked.
I agree with the `BaseException` below in this case.